zipline/data/bundles/alpha_vantage_api.py

```diff
@@ -0,0 +1,239 @@
+import collections
+
+import math
+from datetime import datetime, timedelta, time as dtime
+import numpy as np
+import pandas as pd
+import pickle
+import pytz
+from dateutil import tz
+from trading_calendars import TradingCalendar
+from zipline.data.bundles import core as bundles
+from zipline.data.bundles.universe import Universe, all_alpaca_assets, get_sp500, get_sp100, get_nasdaq100
+from dateutil.parser import parse as date_parse
+
+# uncomment this line if you want a static list of symbols
+#ASSETS = ['AAPL']
+
+ASSETS = None
+UNIVERSE = Universe.NASDAQ100
+
+def list_assets():
+    global ASSETS, UNIVERSE
+    if not ASSETS:
+        if UNIVERSE == Universe.SP100:
+            ASSETS = get_sp100()
+        elif UNIVERSE == Universe.SP500:
+            ASSETS = get_sp500()
+        elif UNIVERSE == Universe.NASDAQ100:
+            ASSETS = get_nasdaq100()
+        else:
+            ASSETS = ['AAPL']
+
+    return sorted(list(set(ASSETS)))
+
+def av_get_data_for_symbol(symbol, start, end, interval):
+    from alpha_vantage.timeseries import TimeSeries
+    import os
+
+    api_key = os.environ['AV_API_KEY']
+
+    ts = TimeSeries(key=api_key)
+    # Get json object with the intraday data and another with  the call's metadata
+
+    if interval == '1m':
+      data, meta_data = ts.get_intraday(symbol, interval='1min', outputsize='full')
+    else:
+      data, meta_data = ts.get_daily_adjusted(symbol, outputsize='full')
+
+    df = pd.DataFrame.from_dict(data, orient='index')
+    df.index = pd.to_datetime(df.index).tz_localize('UTC')
+    
+    df.rename(columns={
+      '1. open': 'open', 
+      '2. high': 'high', 
+      '3. low': 'low', 
+      '4. close': 'close', 
+      '5. volume': 'volume',
+      '6. volume': 'volume',
+      '7. dividend amount': 'dividend', 
+      '8. split coefficient': 'split'
+      }, inplace=True)
+   
+
+    df['open'] = pd.to_numeric(df['open'], downcast='float') 
+    df['high'] = pd.to_numeric(df['high'], downcast='float') 
+    df['low'] = pd.to_numeric(df['low'], downcast='float') 
+    df['close'] = pd.to_numeric(df['close'], downcast='float') 
+    df['volume'] = pd.to_numeric(df['volume'], downcast='unsigned') 
+
+    if 'dividend' in df.columns:
+      df['dividend'] = pd.to_numeric(df['dividend'], downcast='float') 
+
+    if 'split' in df.columns:
+      df['split'] = pd.to_numeric(df['split'], downcast='float') 
+
+    if '5. adjusted close' in df.columns:
+      df.drop('5. adjusted close', inplace=True, axis=1)
```

Comment:
don't know why but it doesnt really work when i try to save the column as 'adj_close'. then it crashes with: `TypeError: Cannot do inplace boolean setting on mixed-types with a non np.nan value`
i don't think it's needed because adjustments get properly written, backtests work fine like this at least

Reply:
did you verify that prices get adjusted during backtests?

Reply:
i tried again and was able to add that column too to db, but it seems to be ignored. if you request historical data or spot-price it will show the unadjusted value. however, calculation for portfolio-value is correct, so in my opinion that's exaclty what we're looking for. what's you thoughts on this?

Reply:
so I'm thinking when you live trade and we have a split - the price is unadjusted.. right? it could be 500$ one day, and be 100$ the next day so that does make sense. it should be reflected in the amount of share though. so if we have 100 shares, we should have 500 the next day (without any new orders)